api/utils/helpers.py

```diff
@@ -25,9 +34,36 @@ def return_short_url(original_url, db_session):
 			return {"error": "Failed to connect to the specified URL"}
 		short_url = generate_short_url(original_url, timestamp)
 		short_url_obj = ShortUrls(
-			original_url = original_url, short_url = short_url)
+			original_url=original_url, short_url=short_url)
 		db_session.add(short_url_obj)
 		db_session.commit()
 		return short_url
 	except Exception as err:
-		return{"error": f"error in processing shortened url"}
\ No newline at end of file
+		return {"error": f"error in processing shortened url"}
+
+
+def is_domain_allowed(original_url, db_session):
+	"""is_domain_allowed function determines if a passed in url is in the allowed list of government
+	   domains. It queries the table allowed_domains to determine if the url is allowed to be shortened
+	   parameter original_url: the url that the user passes to the api
+	   parameter db_session: the database sesion
+	   returns: True if the url contains a domain that is allowed or False if it does not
+	   """
+	try:
+		# Obtain the domain from the url
+		domain = urlparse(original_url).hostname
```

Comment:
I randomly thought of this this morning. I think url parse is returning the hostname. Not necessarily the domain name. So we'd need entries for Canada.ca, nrc.canada.ca etc. Could be problematic with more departments moving to Canada.ca subdomains.

This is potentially further complicated by various other TLDs. https://en.wikipedia.org/wiki/List_of_Internet_top-level_domains

A quick google suggested something like https://pypi.org/project/tldextract/ might be useful.

Reply:
Good point! I have implemented this using urlib and some parsing.